api/handler/v1/guardian_service.go

```diff
@@ -0,0 +1,445 @@
+package v1
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	pb "github.com/odpf/guardian/api/proto/guardian"
+	"github.com/odpf/guardian/appeal"
+	"github.com/odpf/guardian/domain"
+	"github.com/odpf/guardian/policy"
+	"github.com/odpf/guardian/resource"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+type ProtoAdapter interface {
```

Comment:
would it be simpler if we use struct instead of interface for the adapter?

and to make it even more simpler, we could also remove the adapter and change these methods as a standalone functions, just like helper functions. 

since the adapter does not have any state and we only use it in v1 package, I think it should be doable. wdyt?
e.g.
```
package v1

func fromProviderProto(*pb.Provider) (*domain.Provider, error) {
  return
}
```

Reply:
Using interface so that we can create tests for both adapter and grpc service independently. This interface will help to create tests easier by using mock